### PR TITLE
- [mpir] enable M1 builds

### PR DESCRIPTION
**Pull request by SSE4**
closes: #8772

Specify library name and version:  **mpir/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mpir/all/conanfile.py [hunk 1]
```diff
@@ -57,7 +58,7 @@ def build_requirements(self):
             if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
                 self.build_requires("msys2/cci.latest")
 
-    def validate(self):
+    def validate2(self):
```
Comment (jgsogo) [inline recipes/mpir/all/conanfile.py:61]: Just remove if it is not needed.

**Review by jgsogo — CHANGES_REQUESTED** [review]

**Comment by jgsogo** [comment]
... love to see when a PR adds configurations 🚀

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
💯

File: recipes/mpir/all/conanfile.py [hunk 2]
```diff
@@ -101,6 +98,22 @@ def _build_visual_studio(self):
...
+    def _build_context(self):
+        if self.settings.compiler == "apple-clang":
+            env_build = {"CC": tools.XCRun(self.settings).cc,
+                         "CXX": tools.XCRun(self.settings).cxx}
+            if hasattr(self, "settings_build"):
+                # there is no CFLAGS_FOR_BUILD/CXXFLAGS_FOR_BUILD
+                xcrun = tools.XCRun(self.settings_build)
+                flags = " -Wno-implicit-function-declaration -isysroot {} -arch {}".format(xcrun.sdk_path, tools.to_apple_arch(self.settings_build.arch))
+                env_build["CC_FOR_BUILD"] = xcrun.cc + flags
+                env_build["CXX_FOR_BUILD"] = xcrun.cxx + flags
```
Comment (SpaceIm) [inline recipes/mpir/all/conanfile.py:111]: Wouldn't it be more elegant and robust to get `self._autotools.flags` and `self._autotools.cxxflags` in `_configure_autotools()`, then inject those values in `CC_FOR_BUILD` and `CXX_FOR_BUILD` if cross-building (I guess those values are required for some reason in this case)?

**Review by uilianries — APPROVED** [review]

**Pull request merged**